### PR TITLE
Optional pull secret

### PR DESCRIPTION
**Pull request by prary**
Dropping PR #2897 
pull secret should be optional
Resolving issue #2888

### CODE REVIEW

File: pkg/skaffold/build/cluster/sources/sources.go [hunk 1]
```diff
@@ -82,28 +82,29 @@ func podTemplate(clusterDetails *latest.ClusterDetails, artifact *latest.KanikoA
...
-				VolumeSource: v1.VolumeSource{
-					Secret: &v1.SecretVolumeSource{
-						SecretName: clusterDetails.PullSecretName,
-					},
-				},
-			},
-			},
 		},
 	}
+	if clusterDetails.PullSecretName != "" {
```
Comment (tejal29) [inline pkg/skaffold/build/cluster/sources/sources.go]: this function is becoming really long, can we refactor this into 

```suggestion+15
	if clusterDetails.PullSecretName != "" {
	  vms, vs := getVolumeMountsAndVolumes()
	  pod.Spec.Containers[0].VolumeMounts = vms
	  pod.Spec.Volumes = vs  
```

**Comment by priyawadhwa** [comment]
@prary do you mind if I add a test to this PR?

File: pkg/skaffold/build/cluster/sources/sources.go (hunk 1 above)
Comment (prary) [reply pkg/skaffold/build/cluster/sources/sources.go]: ya sure

**Comment by prary** [comment]
@priyawadhwa that would be great :D

**Comment by priyawadhwa** [comment]
Looks like current tests actually cover these changes! I've just refactored the code a little and it should be good to go.

**Pull request merged**